apps/examples/testcase/le_tc/kernel/tc_libc_libgen.c

```diff
@@ -41,31 +41,32 @@
  */
 static void tc_libc_libgen_basename(void)
 {
-	char *psz_str = NULL;
-	char *path = "/etc/passwd";
+	char *path;
+	char *ret_chk;
 
-	psz_str = basename(path);
-	TC_ASSERT_EQ("basename", strncmp(psz_str, "passwd", strlen(psz_str)), 0);
+	path = NULL;
+	ret_chk = basename(path);
+	TC_ASSERT_EQ("basename", strncmp(ret_chk, ".", strlen(".")), 0);
 
-	path = "/etc";
-	psz_str = basename(path);
-	TC_ASSERT_EQ("basename", strncmp(psz_str, "etc", strlen(psz_str)), 0);
-
-	path = "etc";
-	psz_str = basename(path);
-	TC_ASSERT_EQ("basename", strncmp(psz_str, "etc", strlen(psz_str)), 0);
+	path = "\0";
+	ret_chk = basename(path);
+	TC_ASSERT_EQ("basename", strncmp(ret_chk, ".", strlen(".")), 0);
 
 	path = "/";
-	psz_str = basename(path);
-	TC_ASSERT_EQ("basename", strncmp(psz_str, "/", strlen(psz_str)), 0);
+	ret_chk = basename(path);
+	TC_ASSERT_EQ("basename", strncmp(ret_chk, "/", strlen("/")), 0);
+
+	path = "//";
+	ret_chk = basename(path);
+	TC_ASSERT_EQ("basename", strncmp(ret_chk, "/", strlen("/")), 0);
 
-	path = ".";
-	psz_str = basename(path);
-	TC_ASSERT_EQ("basename", strncmp(psz_str, ".", strlen(psz_str)), 0);
+	path = "/dirname/basename";
+	ret_chk = basename(path);
```

Comment:
The base name function tries to modify the constant string.  This should not be allowed.  The test case should raise an data abort.  Please test the same scenario on host PC. This will raise data abort. If it works in tizenRT then it is not acceptable